src/Simplify.cpp

```diff
@@ -4239,6 +4239,53 @@ class Simplify : public IRMutator {
             } else {
                 expr = op;
             }
+        } else if (op->is_intrinsic(Call::prefetch)) {
+            // Collapse the prefetched region into lower dimension whenever is possible.
+            // TODO(psuriana): Deal with negative strides and overlaps.
+
+            vector<Expr> args(op->args);
+            bool changed = false;
+            for (size_t i = 0; i < op->args.size(); ++i) {
+                args[i] = mutate(op->args[i]);
+                if (!args[i].same_as(op->args[i])) {
+                    changed = true;
+                }
+            }
+
+            for (size_t i = 1; i < args.size(); i += 2) {
+                Expr extent_0 = args[i];
+                Expr stride_0 = args[i + 1];
+                for (size_t j = 1; j < args.size(); j+= 2) {
+                    if (i == j) {
+                        continue;
+                    }
+                    Expr extent_1 = args[j];
+                    Expr stride_1 = args[j + 1];
+
+                    if (can_prove(extent_0 * stride_0 == stride_1) || can_prove(extent_1 * stride_1 == stride_0)) {
```

Comment:
Yes in the old version, no in the newest version since the prefetch call args are now sorted based on the storage ordering. I'll change this.